pkg/testrunner/template/helper.go

```diff
@@ -15,23 +15,47 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
-func getLatestK8sVersion(gardenerKubeconfigPath, cloudprofile, cloudprovider string) (string, error) {
+// getVersions returns all K8s version that should be rendered by the chart
+func getVersions(parameters *TestrunParameters) ([]string, error) {
+	if !parameters.MakeVersionMatrix && parameters.K8sVersion != "" {
+		return []string{parameters.K8sVersion}, nil
+	}
+	// if the kubernetes version is not set, get the latest version defined by the cloudprofile
+	if !parameters.MakeVersionMatrix && parameters.K8sVersion == "" {
+		version, err := getLatestK8sVersion(parameters.GardenKubeconfigPath, parameters.Cloudprofile, parameters.Cloudprovider)
+		if err != nil {
+			return nil, fmt.Errorf("Kubernetes is not defined nor can it be read from the cloudprofile: %s", err.Error())
```

Comment:
"Kubernetes is not defined...": what does that mean?
Maybe "`k8s-version` is not defined..."

Reply:
done